xir/validator.py

```diff
@@ -363,6 +363,14 @@ def _shared_definition_checks(
             msg = f"Definition '{name}' is invalid. Wire and parameter names must differ."
             self._validation_messages.append(msg)
 
+        constants_declared = {p for p in declared_params if p in self._program.constants}
```

Comment:
Was a bit too quick on commiting this one. `set(declared_params) - set(self._program.constants)` isn't the same as `{p for p in declared_params if p in self._program.constants}`, since the latter is the intersection between both sets (`set(declared_params).intersection(self._program.constants)`) rather than the difference.

Reply:
Ah, I see! In that case, could we use
```python
constants_declared = set(declared_params) & set(self._program.constants)
```

Reply:
Nice, I didn't know that. Sure, I'll use that instead then. 🎉

Reply:
Is there a preferred way of doing this? I feel like `set(...).intersection(...)` and `set(...) & set(...)` are practically identical, or is there a difference?